src/key_variants.rs

```diff
@@ -4,5 +4,32 @@
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
+use dusk_bls12_381::BlsScalar;
+use dusk_jubjub::JubJubScalar;
+
 pub mod double_key;
 pub mod single_key;
+
+/// Truncate a BLS Scalar to jubjub for safe conversion
+pub fn truncate_bls_to_jubjub(c: BlsScalar) -> JubJubScalar {
+    // Let the bitmask size be `m`
+    // Considering the field size of jubjub is 251 bits, `m < 251`
+    // Plonk logical gates will accept only even `m + 1`, so `(m + 1) % 2 == 0`
+    //
+    // Plonk logical gates will perform the operation from the base bls `r` of
+    // 255 bits + 1. `d = r + 1 - (m + 1) = 4`. But, `d = 4` don't respect the
+    // previously set constraint, so it must be 6.
+    //
+    // This way, the scalar will be truncated to `m = r - d = 255 - 6 = 249
+    // bits`
+    //
+    // The constant below is equivalent to 2^250 - 1
+    let c = c & BlsScalar([
+        0x432667a3f7cfca74,
+        0x7905486e121a84be,
+        0x19c02884cfe90d12,
+        0xa62ffba6a1323be,
+    ]);
+
+    JubJubScalar::from_raw(c.reduce().0)
+}
```

Comment:
This should be implemented in jubjub. Not here. At least IMVHO.

We should expose this safe API there and maybe also provide the truncation solution.

Reply:
I think we all agree this function should be ported somewhere else. Most likely to `jubjub` but it's a bit tricky. The point is, as far as I understood – and it could be wrong – the truncation is made solely due to Plonk constraints. I'm referring specifically to the truncation using the `&`. To be clear, the truncation won't be needed if `Plonk` wasn't involved, the `from_raw(c.reduce().0)` would suffice – or, in the worst case scenario, we would have truncated by JubJub's max value.

So, taking this unto account, I would suggest first of all to make ` BlsScalar([0x432667a3f7cfca74,… ])` as constant. If it's plonk related as it seems to me, it should be a Plonk's constant, so move into that repo.
And then I would just truncate for that constant the number in the function `challenge_hash`.

If instead we want to have a function than "convert bls scalar to jubjub scalar truncating in a way that are plonk friendly" then such function should be still in Plonk IMVHO.

Reply:
As discussed during the stand up we decided to land this PR as is, and move the number as Plonk's constant in the near future, and remove the `truncate_bls_to_jubjub` function from schnorr, until other use cases will rise (e.g. in zedger).